Deployment/src/APIM/Modules/ExternalNotifcationService/main.tf

```diff
@@ -0,0 +1,212 @@
+﻿data "azurerm_api_management" "apim_instance" {
+  resource_group_name = var.resource_group_name
+  name                = var.apim_service_name
+}
+
+# Create apim group
+resource "azurerm_api_management_group" "ens_management_group" {
+  resource_group_name = data.azurerm_api_management.apim_instance.resource_group_name
+  api_management_name = data.azurerm_api_management.apim_instance.name
+  name                = "ens-group-${local.formatted_env}"
+  display_name        = "External Notification Service Group ${var.env_suffix}"
+  description         = "Management group for users with access to the ${var.env} External Notifcation Service."
+}
+
+# Create ENS API
+resource "azurerm_api_management_api" "ens_api" {
+  resource_group_name = data.azurerm_api_management.apim_instance.resource_group_name
+  api_management_name = data.azurerm_api_management.apim_instance.name 
+  name                = "ens-api-${local.formatted_env}"
+  display_name        = "External Notification Service API ${var.env_suffix}"
+  description         = "The External Notification Service API provides the ability to subscribe and deliver notifications."
+  revision            = "1"
+  path                = "ens-api-${local.formatted_env}"
+  protocols           = ["https"]
+  service_url         = var.apim_api_service_url
+
+  subscription_key_parameter_names {
+    header = "Ocp-Apim-Subscription-Key"
+    query  = "subscription-key"
+  }
+
+  import {
+    content_format = "openapi"
+    content_value  = var.api_openapi_spec_path
+  }
+}
+
+# Create D365 Product
+resource "azurerm_api_management_product" "d365_product" {
+  resource_group_name   = data.azurerm_api_management.apim_instance.resource_group_name
+  api_management_name   = data.azurerm_api_management.apim_instance.name
+  product_id            = "ens-d365-${local.formatted_env}"
+  display_name          = "External Notification Service for D365 ${var.env_suffix}"
+  description           = "The External Notification Service provides the ability to subscribe and deliver notifications and to be used by D365."
+  subscription_required = true
+  approval_required     = true
```

Comment:
I believe this means 'requires admin approval' i.e. needs an admin to manually accept or reject subs. Is this as intended?